x/posts/legacy/v0.13.0/types.go

```diff
@@ -0,0 +1,40 @@
+package v0130
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v0100 "github.com/desmos-labs/desmos/x/posts/legacy/v0.10.0"
+	v0120 "github.com/desmos-labs/desmos/x/posts/legacy/v0.12.0"
+	v040posts "github.com/desmos-labs/desmos/x/posts/legacy/v0.4.0"
+	v060 "github.com/desmos-labs/desmos/x/posts/legacy/v0.6.0"
+	v080posts "github.com/desmos-labs/desmos/x/posts/legacy/v0.8.0"
+)
+
+// GenesisState contains the data of a v0.12.0 genesis state for the posts module
+type GenesisState struct {
+	Posts               []Post                            `json:"posts"`
+	UsersPollAnswers    map[string][]v040posts.UserAnswer `json:"users_poll_answers"`
+	PostReactions       map[string][]v060.PostReaction    `json:"post_reactions"`
+	RegisteredReactions []v040posts.Reaction              `json:"registered_reactions"`
+	Params              v080posts.Params                  `json:"params"`
+}
+
+type OptionalData struct {
```

Comment:
I think it would be better to rename this an `OptionalDataEntry` instead, since it represents a single entry inside the optional data array